### PR TITLE
Feat/Add gym selection page

### PR DESCRIPTION
**Pull request by yarkhinephyo**
#### Summary
The users can select gyms to see where they are. (But cannot look at the map to see all gyms)

#### Subsequent PRs
Clicking on search routes will display the routes instead of the map.
Clicking on request gym will lead to another page where user can fill in form.

image

### CODE REVIEW

File: ionic_user_interface/src/components/gym-selector/AutoComplete.vue [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    return {
+      ...toRefs(state),
+      onSelected,
+      onEnter,
+    };
+  },
+});
+</script>
+
+<!-- Add "scoped" attribute to limit CSS to this component only -->
```
Comment (qwoprocks) [inline ionic_user_interface/src/components/gym-selector/AutoComplete.vue:96]: lmao is this comment even needed

File: ionic_user_interface/src/components/gym-selector/GymSelector.vue [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+      clickCloseErrorMsg,
+      closeCircleOutline,
+      onClickSearchRoutes,
+      onClickCantFindGym,
+    };
+  },
+});
+</script>
+
+<style></style>
```
Comment (qwoprocks) [inline ionic_user_interface/src/components/gym-selector/GymSelector.vue:173]: Should just remove?

File: ionic_user_interface/src/composables/useCanvasWidth.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import { Ref, ref } from 'vue';
+import { debounce } from 'lodash';
+
+export function useCanvasWidth(): {
+  canvasWidth: Ref<number>;
+  updateCanvasWidth: () => void;
+} {
+  const canvasWidth = ref(0);
```
Comment (qwoprocks) [inline ionic_user_interface/src/composables/useCanvasWidth.ts:8]: Maybe we should set a default of 700px or something

File: ionic_user_interface/src/components/gym-selector/AutoComplete.vue [hunk 4]
```diff
@@ -0,0 +1,101 @@
+<template>
```
Comment (qwoprocks) [inline ionic_user_interface/src/components/gym-selector/AutoComplete.vue:1]: Can we implement tab-autocomplete?

**Review by qwoprocks — APPROVED** [review]

**Comment by qwoprocks** [comment]
Also should reset the map + gym list reactively when the country changes

File: ionic_user_interface/src/components/gym-selector/AutoComplete.vue (hunk 4 above)
Comment (yarkhinephyo) [reply ionic_user_interface/src/components/gym-selector/AutoComplete.vue:1]: what a programmer

File: ionic_user_interface/src/composables/useCanvasWidth.ts (hunk 3 above)
Comment (yarkhinephyo) [reply ionic_user_interface/src/composables/useCanvasWidth.ts:8]: I think this is fine because they won't be shown at all if empty

**Comment by yarkhinephyo** [comment]
> Also should reset the map + gym list reactively when the country changes

Done

**Pull request merged**